### ISSUE TITLE
Add support for patternProperties

### ISSUE THREAD

**Issue by Darkhogg** [willfarrell/apidoc-plugin-schema#11]
The `patternProperties` keyword is completely ignored.  It should appear in the list of params as any other property.

**Comment by willfarrell** [comment]
I'd be more than happy to merge a PR for this feature.

**Comment by Darkhogg** [comment]
I managed to add it ad-hoc for a single run, I'll send a PR with a proper solution if I have some time.

My idea was to add an argument called "PATTERN_#" with increasing # for different patterns and specify the pattern in the description, kind of like this:
![patternPrperties PoC](https://user-images.githubusercontent.com/237337/37350549-57c9df74-26d9-11e8-9c8a-574768a524d1.png)

let me know if that's ok or you have another idea in mind, and I'll see what I can do.

**Comment by willfarrell** [comment]
I think I'd rather have the pattern itself as the name

**Comment by Darkhogg** [comment]
Yeah, I'd rather too, but apiDoc does very strange things if the name is not alphanumeric, and would straight up not work if `.` is in the pattern. At least that has been my experience, I'm all ears if there's a solution.
